kohttp/src/main/kotlin/io/github/rybalkinsd/kohttp/client/ClientBuilderImpl.kt

```diff
@@ -80,21 +77,17 @@ internal class ClientBuilderImpl : ClientBuilder {
         set(value) { builder.cache(value) }
         get() = throw UnsupportedOperationException()
 
-    override var socketFactory: SocketFactory
-        set(value) { builder.socketFactory(value) }
-        get() = throw UnsupportedOperationException()
-
-    override var sslSocketFactory: SSLSocketFactory
-        @Suppress("DEPRECATION")
-        set(value) { builder.sslSocketFactory(value) }
-        get() = throw UnsupportedOperationException()
-
-    override var hostnameVerifier: HostnameVerifier
-        set(value) { builder.hostnameVerifier(value) }
-        get() = throw UnsupportedOperationException()
+    override var sslConfig: SslConfig
+        set(value) {
+            if (value.sslSocketFactory != null && value.trustManager != null) {
+                builder.sslSocketFactory(value.sslSocketFactory!!, value.trustManager!!)
```

Comment:
Smartcast here should take care of nullability here
```suggestion
                builder.sslSocketFactory(value.sslSocketFactory, value.trustManager)
```

Reply:
By removing !! I am getting the following warning: `w: /kohttp/kohttp/src/main/kotlin/io/github/rybalkinsd/kohttp/client/ClientBuilderImpl.kt: (83, 42): Type mismatch: inferred type is SSLSocketFactory? but SSLSocketFactory was expected`